pkg/utils/k8sutil.go

```diff
@@ -376,13 +375,7 @@ func EnsureK8sResources(ns, name string, funcObj *spec.Function, client kubernet
 	}
 
 	// update deployment for memory request
-	if funcObj.Spec.Memory != (resource.Quantity{}) {
-		resource := map[v1.ResourceName]resource.Quantity{
-			v1.ResourceMemory: funcObj.Spec.Memory,
-		}
-		dpm.Spec.Template.Spec.Containers[0].Resources.Requests = resource
-		dpm.Spec.Template.Spec.Containers[0].Resources.Limits = resource
-	}
+	dpm.Spec.Template.Spec.Containers[0].Resources = funcObj.Spec.Template.Spec.Containers[0].Resources
```

Comment:
Just copy the entire template as your starting point, rather than selectively supporting only certain values:

```go
import "k8s.io/client-go/pkg/api"

tmplCopy, err := api.Scheme.DeepCopy(funcObj.Spec.Template)
if err != nil {
   return err
}
dpm.Spec.Template = tmplCopy.(*metav1.PodTemplateSpec)
```

Then you can go and override parts of dpm.Spec, like Containers[0].Image, etc.